src/Collections-Arithmetic/Collection.extension.st

```diff
@@ -108,6 +108,11 @@ Collection >> floor [
 	^ self collect: [:a | a floor]
 ]
 
+{ #category : #'*Collections-arithmetic-collectors' }
+Collection >> isZero [
```

Comment:
We are reviewing this issue with Stef and we don't like to pollute collections with this kind of interface.

We were playing a bit around and it would seem that replacing `SmallInteger >> #/`with:

```smalltalk
/ aNumber 
	"Primitive. This primitive (for /) divides the receiver by the argument
	and returns the result if the division is exact. Fail if the result is not a
	whole integer. Fail if the argument is 0 or is not a SmallInteger. Optional.
	No Lookup. See Object documentation whatIsAPrimitive."

	<primitive: 10>
	aNumber = 0 ifTrue: [^(ZeroDivide dividend: self) signal].
	^(aNumber isMemberOf: SmallInteger)
		ifTrue: [(Fraction numerator: self denominator: aNumber) reduced]
		ifFalse: [super / aNumber]
```

This checks zero directly with `= 0`, meaning this will only work with numbers, but in case the denominator is not a number then it will fall back to the superclass method that eventually makes the double dispatch with `adaptToNumber:andSend:`. This will probably make the slower case of a zero division a bit slower, but it's already slow since the exception will force the reification of the stack...

On the other side, this change removes some pressure on the API, we do not need to implement `isZero` in all possible denominators, just `adaptToNumber:andSend:`.